Assets/Scripts/Dialogue System/Dialogue Editor/GraphSaveUtility.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Experimental.GraphView;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+
+//lets graphees be saved and loaded into editor
+public class GraphSaveUtility 
+{
+    private DialogueGraphView _targetGraphView;
+    private DialogueContainer _containerCache;
+
+    private List<Edge> Edges => _targetGraphView.edges.ToList();
+    private List<DialogueNode> Nodes => _targetGraphView.nodes.ToList().Cast<DialogueNode>().ToList();
+
+
+    public static GraphSaveUtility GetInstance(DialogueGraphView targetGraphView) {
+        return new GraphSaveUtility{
+            _targetGraphView = targetGraphView
+            }; 
+        }
+    public void SaveGraph(string fileName) {
+        //dont need to save empty graph
+        if(!Edges.Any())
+                return;
+        var dialogueContainer = ScriptableObject.CreateInstance<DialogueContainer>();
+        var connectedPorts = Edges.Where(x => x.input.node != null).ToArray(); //only count connected edges
+        for(int i = 0; i < connectedPorts.Length; i++) {
+            var outputNode = connectedPorts[i].output.node as DialogueNode;
+            var inputNode = connectedPorts[i].input.node as DialogueNode;
+
+            dialogueContainer.NodeLinks.Add(new NodeLinkData {
+                BaseNodeGuID = outputNode.GUID,
+                PortName = connectedPorts[i].output.portName,
+                TargetNodeGuID = inputNode.GUID
+                }) ;
+            }
+        foreach(var dialgoueNode in Nodes.Where(node => !node.EntryPoint)) { //dont need start node
+            dialogueContainer.DialogueNodeData.Add(new DialogueNodeData{
+                NodeGuID = dialgoueNode.GUID,
+                DialogueText = dialgoueNode.DialogueText,
+                position = dialgoueNode.GetPosition().position
+                });
+            }
+
+        //create folder if necessary
+        if (!AssetDatabase.IsValidFolder("Assets/Resources")) {
+            AssetDatabase.CreateFolder("Assets", "Resources");
+            }
+        //save graph
+        AssetDatabase.CreateAsset(dialogueContainer, $"Assets/Resources/{fileName}.asset");
+        AssetDatabase.SaveAssets();
```

Comment:
might be a good idea to refactor save location to a single string at top of file that gets referenced to make changing it easier